lib/supavisor/helpers.ex

```diff
@@ -174,4 +261,45 @@ defmodule Supavisor.Helpers do
 
     {[client_final_without_proof, ",p=", Base.encode64(client_proof)], server_signature}
   end
+
+  def get_client_final(
+        :auth_query,
+        secrets,
+        srv_first,
+        client_nonce,
+        user_name,
+        channel
+      ) do
+    channel_binding = "c=#{channel}"
+    nonce = ["r=", srv_first[:nonce]]
+
+    client_first_bare = [<<"n=">>, user_name, <<",r=">>, client_nonce]
+    server_first = srv_first[:raw]
```

Comment:
```suggestion
    nonce = ["r=", srv_first.nonce]

    client_first_bare = [<<"n=">>, user_name, <<",r=">>, client_nonce]
    server_first = srv_first.raw
```

So we raise if those keys are missing? Otherwise we fail with a `nil` somewhere...